pandas/_libs/src/ujson/python/objToJSON.c

```diff
@@ -1782,7 +1765,7 @@ void Object_beginTypeContext(JSOBJ _obj, JSONTypeContext *tc) {
         pc->PyTypeToJSON = PyUnicodeToUTF8;
         tc->type = JT_UTF8;
         return;
-    } else if (PyObject_IsInstance(obj, type_decimal)) {
```

Comment:
are there any Decimal subclasses to worry about?

Reply:
PyObject_TypeCheck returns true for subtypes as well, so I don't think that matters:

https://docs.python.org/3.7/c-api/object.html#c.PyObject_TypeCheck